samtranslator/model/api/api_generator.py

```diff
@@ -204,20 +206,81 @@ def _construct_stage(self, deployment, swagger):
         stage.TracingEnabled = self.tracing_enabled
 
         if swagger is not None:
-            deployment.make_auto_deployable(stage, self.remove_extra_stage, swagger)
+            deployment.make_auto_deployable(stage, self.remove_extra_stage, swagger, self.domain)
 
         if self.tags is not None:
             stage.Tags = get_tag_list(self.tags)
 
         return stage
 
+    def _construct_api_domain(self, rest_api):
+        """
+        Constructs and returns the ApiGateway Domain and BasepathMapping
+        """
+        if self.domain is None or \
+           self.domain.get('DomainName') is None or \
+           self.domain.get('CertificateArn') is None:
+            return None, None
+
+        logical_id = logical_id_generator.LogicalIdGenerator("", self.domain).gen()
+
+        domain = ApiGatewayDomainName('ApiGatewayDomainName' + logical_id,
+                                      attributes=self.passthrough_resource_attributes)
+        domain.DomainName = self.domain.get('DomainName')
+        endpoint = self.domain.get('EndpointConfiguration')
+
+        if endpoint is None or endpoint not in ['EDGE', 'REGIONAL']:
```

Comment:
Again, should we be silently falling back to default values? What if they use an intrinsic function in EndpointConfiguration? I don't think it's fair to say we support all intrinsics for the Domain property. What if I wanted to do something like this?

```yaml
    EndpointConfiguration: !If
      - SomeCondition
      - REGIONAL
      - EDGE
```

Reply:
Will add a test

Reply:
I see, that makes sense. I've definitely been bitten by SAM lacking support for `!If` on certain parameters so I'd like to see that supported here, ideally. I'm fine with having EndpointConfiguration be optional defaulting to REGIONAL. However, I'm not ok with us defaulting to REGIONAL if someone puts `EndpointConfiguration: INVALID` in their template. I'd rather we throw an error in that case.